lib/Associations/One.js

```diff
@@ -214,52 +215,39 @@ function extendInstance(Model, Instance, Driver, association) {
     writable: true
   });
   Object.defineProperty(Instance, association.setAccessor, {
-    value: function (OtherInstance, cb) {
+    value: function (OtherInstance, next) {
       if (association.reversed) {
         Instance.save(function (err) {
           if (err) {
-            return cb(err);
+            return next(err);
           }
 
           if (!Array.isArray(OtherInstance)) {
             util.populateConditions(Model, Object.keys(association.field), Instance, OtherInstance, true);
 
-            return OtherInstance.save({}, { saveAssociations: false }, cb);
+            return OtherInstance.save({}, { saveAssociations: false }, next);
           }
 
-          var associations = _.clone(OtherInstance);
-
-          var saveNext = function () {
-            if (!associations.length) {
-              return cb();
-            }
-
-            var other = associations.pop();
```

Comment:
I also changed the order in which `hasOne` associations are saved from reverse to provided order.
No tests for this as it doesn't really affect anything.